cpp/src/parquet/properties.h

```diff
@@ -843,6 +863,9 @@ class PARQUET_EXPORT ArrowWriterProperties {
     bool store_schema_;
     bool compliant_nested_types_;
     EngineVersion engine_version_;
+
+    bool use_threads_;
+    ::arrow::internal::Executor* executor_;
```

Comment:
Is `executor_` guaranteed to be initialized?

Reply:
No, the default value is nullptr and executor created by `::arrow::internal::GetCpuThreadPool()` will be used by default. If this is not a good idea, I can set the default value to `::arrow::internal::GetCpuThreadPool()` to make follow-up use much simpler.

Reply:
When is it default nullified? I didn't see related code.

Reply:
I have changed to return `::arrow::internal::GetCpuThreadPool()` if user does not provide a valid executor. Please check the related change in the `properties.cc`

Reply:
```
::arrow::internal::Executor* ArrowWriterProperties::executor() const {
  return executor_ != nullptr ? executor_ : ::arrow::internal::GetCpuThreadPool();
}
```